src/components/Header.jsx

```diff
@@ -1,25 +1,28 @@
 import React from 'react';
 import BaseComponent from 'lib/BaseComponent';
-import { Link } from 'react-router';
+import { Link, browserHistory } from 'react-router';
 
 export default class Header extends BaseComponent {
+  // Allows user to navigate to populated search page on pressing 'Enter'
+  handleKeyPress (e) {
+    if (e.nativeEvent.key == 'Enter'){ // Enter pressed
+      browserHistory.push('/search?q=' + e.target.value);
```

Comment:
You should do some validation on the value. Maybe lowercase it, strip it of whitespace, etc.